packages/devtools_app/lib/src/charts/flame_chart.dart

```diff
@@ -1035,7 +1149,21 @@ class FlameChartNode<T extends FlameChartDataMixin<T>> {
     required this.onSelected,
     this.selectable = true,
     this.sectionIndex = 0,
-  });
+  }) {
+    final paragraphBuilder = ParagraphBuilder(
+      ParagraphStyle(
+        ellipsis: '...',
+        textAlign: TextAlign.left,
+        maxLines: 1,
+        fontStyle: FontStyle.normal,
+        textDirection: TextDirection.ltr,
+        fontSize: 10,
+      ),
+    );
+    paragraphBuilder.pushStyle(ui.TextStyle(color: _darkTextColor));
+    paragraphBuilder.addText(text);
+    paragraph = paragraphBuilder.build();
```

Comment:
is it more performant to do this as part of the constructor than in draw? I'm assuming because draw is called more than once and the node should only be created once?

Reply:
It might be even more performant to do it lazily, since we start out sort of zoomed out with most text invisible. @dnfield is the expert, but I believe paragraph builders are fairly expensive to construct so re-using them is a big win